main.js

```diff
@@ -392,6 +392,51 @@ if(typeof System !== "undefined") {
 	addTilde(System);
 }
 
+function addContextual(loader){
+  loader._contextualModules = {};
+
+  loader.setContextual = function(moduleName, definer){
+    // `definer` could be a function or could be a moduleName
+		if (typeof definer === 'string') {
+			this._contextualModules[moduleName] = this['import'](definer);
```

Comment:
I was going to put that on the steal-clone PR. Do you want it on this branch instead?